### PR TITLE
Make access to the Capability IStorage less annoying

### PR DESCRIPTION
**Pull request by gigaherz**
This PR adds a couple of methods that allow accessing the storage directly from the capability. This reduces the extremely reduncant calls such as:

```java
NBTBase tag = CapabilityItemHandler.ITEM_HANDLER_CAPABILITY.getStorage().writeNBT(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, inventory, side);
```

to a more manageable:

```java
NBTBase tag = CapabilityItemHandler.ITEM_HANDLER_CAPABILITY.writeNBT(inventory, side);
```

Which I believe turns capability storage from something no one would ever use into an useful storage solution, without changing any of the semantics of the IStorage itself.

### CODE REVIEW

**Comment by williewillus** [comment]
+1 many times over for me.
I want to use IItemhandler's default storage but it's just so verbose .-.

File: src/main/java/net/minecraftforge/common/capabilities/Capability.java [hunk 1]
```diff
@@ -65,10 +65,29 @@
...
     /**
      * @return An instance of the default storage handler. You can safely use this store your default implementation in NBT.
      */
     public IStorage<T> getStorage() { return storage; }
+    
+    /**
+     * Quick access to the IStorage's readNBT. 
+     * See {@link IStorage#readNBT(Capability, Object, EnumFacing, NBTBase)}  for documentation.
+     */
+    public void readNBT(T instance, EnumFacing side, NBTBase nbt)
```
Comment (kashike) [inline src/main/java/net/minecraftforge/common/capabilities/Capability.java]: What about naming this `read`? Having `NBT` in the name is redundant, seeing as the method accepts `NBTBase`?

File: src/main/java/net/minecraftforge/common/capabilities/Capability.java [hunk 2]
```diff
@@ -65,10 +65,29 @@
...
+    {
+    	storage.readNBT(this, instance, side, nbt); 
+    }
+    
+    /**
+     * Quick access to the IStorage's writeNBT. 
+     * See {@link IStorage#writeNBT(Capability, Object, EnumFacing)} for documentation.
+     */
+    public NBTBase writeNBT(T instance, EnumFacing side)
```
Comment (kashike) [inline src/main/java/net/minecraftforge/common/capabilities/Capability.java]: What about naming this `write`? Having `NBT` in the name is redundant, seeing as the method accepts `NBTBase`?

File: src/main/java/net/minecraftforge/common/capabilities/Capability.java (hunk 1 above)
Comment (gigaherz) [inline src/main/java/net/minecraftforge/common/capabilities/Capability.java]: It's consistent with every single NBT method we have everywhere else, though. If we want to change that practice, I'm all up for it, although I wouldn't call it just "read". If I were to remove NBT, I'd rename it to deserialize instead.

File: src/main/java/net/minecraftforge/common/capabilities/Capability.java (hunk 2 above)
Comment (gigaherz) [inline src/main/java/net/minecraftforge/common/capabilities/Capability.java]: Same as above, if we decide to change this, I'll rename to `serialize`.

File: src/main/java/net/minecraftforge/common/capabilities/Capability.java (hunk 1 above)
Comment (williewillus) [inline src/main/java/net/minecraftforge/common/capabilities/Capability.java]: Seeing as it's just a forwarding call, it makes sense to name it the same as what you're forwarding to

**Pull request merged**